src/App.tsx

```diff
@@ -1,61 +1,165 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+
+enum ChangeType {
+  TITLE,
+  USER,
+  COMPLETED,
+}
+
+export const App: React.FC<{}> = () => {
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState(0);
+  const [completed, setCompleted] = useState(false);
+  const [todos, setTodos] = useState(
+    todosFromServer.sort((a, b) => a.id - b.id),
+  );
+
+  const [errors, setErrors] = useState({
+    userNotSelected: false,
+    titleIsEmpty: false,
+  });
+
+  const clearForm = () => {
+    setTitle('');
+    setUser(0);
+    setCompleted(false);
+  };
+
+  const handleSubmit = (
+    e: React.SyntheticEvent | React.MouseEvent<HTMLButtonElement>,
+  ) => {
+    e.preventDefault();
+    setErrors({
+      userNotSelected: user === 0,
+      titleIsEmpty: title.length === 0,
+    });
+
+    setTodos(prev => {
+      if (user === 0 || title.length === 0) {
+        return prev;
+      }
+
+      const newTodo = {
+        id: prev[prev.length - 1].id + 1,
+        title,
+        userId: user,
+        completed,
+      };
+
+      clearForm();
+
+      return ([
+        ...prev,
+        newTodo,
+      ]);
+    });
+  };
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+    field: ChangeType,
+  ) => {
+    switch (field) {
+      case ChangeType.TITLE:
+        setTitle(e.target.value);
+        setErrors(prev => ({
+          userNotSelected: prev.userNotSelected,
+          titleIsEmpty: false,
+        }));
+        break;
+      case ChangeType.USER:
+        setUser(Number.parseInt(e.target.value, 10));
+        setErrors(prev => ({
+          userNotSelected: false,
+          titleIsEmpty: prev.titleIsEmpty,
+        }));
+        break;
+      default: // ChangeType.COMPLETED
+        setCompleted(prev => !prev);
+        break;
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">Title: </label>
+          <input
+            id="title"
+            name="title"
+            type="text"
+            value={title}
+            data-cy="titleInput"
+            onChange={e => handleChange(e, ChangeType.TITLE)}
+            placeholder="Enter title..."
+          />
+          {errors.titleIsEmpty
+          && <span className="error">Please enter a title</span>}
```

Comment:
let's polish it a bit
```suggestion
          {errors.titleIsEmpty && (
            <span className="error">
              Please enter a title
            </span>
           )}
```